### PR TITLE
Minor Eglot improvements

### PR DESCRIPTION
**Pull request by elken**
A couple of small changes that improve the Eglot experience.

### CODE REVIEW

**Comment by aikrahguzar** [comment]
> A couple of small changes that improve the Eglot experience.

I think it would be a good idea to also add `markdwon-mode` to `packages.el` since otherwise eldoc doesn't render docs very well.

**Comment by elken** [comment]
No need, it'll get pulled in if you have the markdown module enabled. It's a weak dependency and docs look fine without it.

image

**Comment by aikrahguzar** [comment]
> No need, it'll get pulled in if you have the markdown module enabled. It's a weak dependency and docs look fine without it.
> 
> image

It is a weak dependency but I think quite important for a good experience. At least for haskell, I will get just
```haskell`
in the first line of docs without markdown-mode.

I use `(setq eldoc-echo-area-use-multiline-p nil)` so that the mini-window is not constantly resized which I find distracting. But without markdown-mode  the only thing I see is the opening delimiter of the haskell block.

By the way, I am seeing some fontification in the docs in your picture which makes me think that the markdown mode is involved in rendering those.

**Comment by elken** [comment]
It's a bare `emacs -Q` so there's nothing involved except eglot & clojure-mode

File: modules/editor/evil/config.el [hunk 1]
```diff
@@ -533,7 +533,11 @@ directives. By default, this only recognizes C directives.")
        (:after elfeed
         :map elfeed-search-mode-map
         :n "gr" #'elfeed-search-update--force
-        :n "gR" #'elfeed-search-fetch))
+        :n "gR" #'elfeed-search-fetch)
+       (:after eglot
+        :map eglot-mode-map
+        :nv "gd" #'+lookup/definition
+        :nv "gD" #'+lookup/references))
```
Comment (regadas) [inline modules/editor/evil/config.el:540]: 🙌

**Comment by hlissner** [comment]
Excellent work, as always! Thanks again!

**Comment by yuanl** [comment]
advice around #'eglot-ensure may have issue here. All [javascript major mode define](https://github.com/joaotavora/eglot/blob/master/eglot.el#L230-L234) has another form like `(MAJOR-MODE-SYMBOL :LANGUAGE-ID ID)` in `eglot-server-programs`.

```lisp
(((js-mode :language-id "javascript")
  (js-ts-mode :language-id "javascript")
  (tsx-ts-mode :language-id "typescriptreact")
  (typescript-ts-mode :language-id "typescript")
  (typescript-mode :language-id "typescript"))
 . ("typescript-language-server" "--stdio"))
```

The adviced function will always return `nil` in js-mode or rjsx-mode which cause eglot-ensure can not be use in javascript. A tide server will started instead.

Deleting this advice is a workaround in my test.

**Comment by hlissner** [comment]
@yuanl Should be fixed in 173842c.

**Pull request merged**